repos/fdbt-reference-data-service/src/uploaders/txc_uploader/txc_processor.py

```diff
@@ -177,45 +186,107 @@ def collect_journey_patterns(data: dict, service: dict):
                 journey_pattern_section_refs, raw_journey_pattern_sections
             ),
             "journey_pattern_info": journey_pattern["journey_pattern_info"],
+            "journey_pattern_section_refs": journey_pattern_section_refs,
         }
         journey_patterns.append(processed_journey_pattern)
 
     return journey_patterns
 
+def check_journey_pattern_exists(
+    cursor,
+    op_service_id,
+    destination_display,
+    direction,
+    logger,
+):
+    query = """
+        SELECT id FROM txcJourneyPatternNew
+        WHERE operatorServiceId <=> %s AND destinationDisplay <=> %s AND direction <=> %s
+        LIMIT 1
+    """
+
+    cursor.execute(
+        query,
+        [
+            op_service_id,
+            destination_display,
+            direction
+        ],
+    )
+    result = cursor.fetchone()
+
+    journey_pattern_id = result[0] if result and len(result) > 0 else None
+    if journey_pattern_id:
+        logger.info(
+            f"Existing journey pattern found - '{op_service_id}' - '{destination_display}' - '{direction}''"
+        )
+
+    return True if journey_pattern_id else False
 
 def iterate_through_journey_patterns_and_run_insert_queries(
-    cursor, data: dict, operator_service_id: str, service: dict
+    cursor,
+    data: dict,
+    operator_service_id: str,
+    service: dict,
+    vehicle_journeys: list,
+    logger,
 ):
     journey_patterns = collect_journey_patterns(data, service)
+
+    vehicle_journey_journey_pattern_refs = [
+        vehicle_journey["journey_pattern_ref"] for vehicle_journey in vehicle_journeys
+    ]
+
     for journey_pattern in journey_patterns:
+        journey_pattern_info = journey_pattern["journey_pattern_info"]
+
+        if (
+            journey_pattern_info["journey_pattern_ref"]
+            not in vehicle_journey_journey_pattern_refs
+        ):
+            continue
+
+        if check_journey_pattern_exists(
```

Comment:
I'd maybe remove this check and the `check_journey_pattern_exists` function, it might lead to some issues since it's not including the section refs like disruptions does

Reply:
done